### PR TITLE
Fix/technical debt dojo finding

### PR DESCRIPTION
**Pull request by renejal**
## Description

Our organization's objective is that the project meets all the quality standards 

The following image shows that we have a technical debt of 58 days and a coverage of 66.2%.

image

- The refactoring of the file pep8 dojo/finding/views.py is performed

* Target is 2 days and over 70 coverage

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -41,3 +44,4 @@ exclude =
     # This contains of branch that we don't want to check
     # dev
 max-complexity = 10
+max-line-length = 120
```
Comment (Maffooch) [inline .flake8:47]: I am nervous this will force a lot of unrelated changes for PRs that are fixing small issues. There are likely a lot of instances of line length > 120

Comment (renejal) [reply .flake8]: It is correct the project has many max-line-lenght problems, with this contribution https://github.com/DefectDojo/django-DefectDojo/pull/8046
we incorporate a pre-commit so that the technical debt does not continue to grow.

In the future we should remove all the ignore in the .flacke8 file.

**Comment by renejal** [comment]
It is correct the project has many max-line-lenght problems, with this contribution https://github.com/DefectDojo/django-DefectDojo/pull/8046
we incorporate a pre-commit so that the technical debt does not continue to grow.

In the future we should remove all the ignore in the .flacke8 file.

File: .flake8 (hunk 1 above)
Comment (Maffooch) [reply .flake8]: It would be ideal to correct all instances of line length violations before enabling this rule. Otherwise, all PRs going forwards will have failing unit tests until the entire project is no longer in violation

**Comment by mtesauro** [comment]
So, there's a lot of work here to get all lines below 120 characters. I did a bit of bash scripting to get an idea of the scope of the problem.

```
$ ./failure-counts.bash 
==[ django-DefectDojo repo stats ]==
 Total number of .py files: 982
   Total failing .py files: 381
```

One thing I would definitely suggest, taking tests (./tests/*) and unittests (./unittests/*) out of scope for pep8:

```
$ grep "Failed file" long-lines.report | wc -l
381
$ grep "Failed file" long-lines.report | grep -v "./tests/" | grep -v "./unittests/" | wc -l
276
$ expr 381 - 276
105
```

They way to read the above is that 
* There are 381 total files that fail pep8 (120+ chars) 
* If you remove the 276 failing files in ./tests/ and ./unittests/ there's only 105 files that fail pep8 (120 chars)

Some other global numbers (aka numbers without tests & unittests filtered out) 

Total number of lines that fail: 3509
```
$ find ./ -type f -name "*.py" -exec ./pep8-line-checker.bash {} \; | grep "will fail pep8" | wc -l
3509
```

The worst offender for long lines is ./unittests/tools/test_anchore_grype_parser.py
```
$ find ./ -type f -name "*.py" -exec ./longest-line.bash {} \; | grep "length of line"| cut -d " " -f 1,8 | sort -u | tail -n 1
728 ./unittests/tools/test_anchore_grype_parser.py
```

I've attached a "full report" of which files have long (120+) lines, what line numbers are too long and total number of failed lines per file.
[long-lines.txt](https://github.com/DefectDojo/django-DefectDojo/files/11392964/long-lines.txt)

File: .flake8 (hunk 1 above)
Comment (renejal) [reply .flake8]: @maffooch and @mtesauro, thank you for your contributions.

As far as I understand, you are running flake8 in the pipeline, but ignoring several errors. At this point, we agree not to modify these rules until we have everything corrected. Not only the maximum number of lines, which is the easiest and can be achieved with black automatically up to a certain point and is already integrated in the PR pre-commit. https://github.com/DefectDojo/django-DefectDojo/pull/8053

Initially, we thought about creating a single PR https://github.com/DefectDojo/django-DefectDojo/pull/8013 with all the corrections and improvements in code readability according to flake8, black, pep8. But we realized that there may be too many changes and it could complicate things for merging.

That's why we launched this PR with the necessary corrections reported by SonarQube for the dojo/finding/views.py module, making it much easier to review and reduce the chances of conflicts with other PRs.

The idea is to continue contributing with small contributions module by module incrementally until technical debt is completely eliminated. But before that, we must avoid it from growing, which is why we propose this pre-commit. It will only run on the modules that were modified and not on the entire project, and it is the developer's responsibility to correct it if necessary. At this stage, it is important to remember that we have black, which saves us a lot of work. I say this because I have already done it, and it didn't take me more than two days to correct the entire project with the proposed pre-commit configuration.

@mtesauro, regarding your recommendation, we totally agree. We are not taking into account (./test) and (./unittest).

In this image, we can see the problems we currently have. Excluding the coverage.

image

Comment (Maffooch) [reply .flake8]: @renejal I think we are on the same page now. I think the course of action should be as follows
1. Close #8013 
2. Update the code base one by one to be in compliance with Black, Pep8, and Flake8
3. Once the code base is in compliance, add the pre commit hooks and update the unit tests to ensure the code base stays in compliance.

**Comment by damiencarol** [comment]
I agree with @mtesauro , scanning `./tests/` or `./unittests/` folders doesn't make sense.

File: .flake8 (hunk 1 above)
Comment (renejal) [reply .flake8]: Perfect

File: dojo/finding/views.py [hunk 2]
```diff
@@ -1542,143 +2024,174 @@ def apply_cwe_mitigation(apply_to_findings, template, update=True):
...
+                extra_tags="alert-danger",
+            )
     else:
         return HttpResponseForbidden()
 
 
-def download_finding_pic(request, token):
+# Remove the unused function parameter "request"
+# TODO : REVISAR
+def download_finding_pic(token):
```
Comment (Maffooch) [inline dojo/finding/views.py:2162]: I believe the request object needs to be there whether it is used or not. Probably a Django requirement. If Pep8/Black is complaining about it not being used, the we could add some check to make sure the path is what is expected and add a comment explaining the reason for the useless check

File: dojo/finding/views.py [hunk 3]
```diff
@@ -1690,113 +2203,151 @@ class Original(ImageSpec):
...
-
-                        if finding.file_path:
-                            file_path = finding.file_path
+                        # "line" is not accessedPylance
+                        # if finding.line:
+                        # line = finding.line
+                        # ------------------
+                        # "file_path is no accesedpylance"
+                        # if finding.file_path:
+                        #     file_path = finding.file_path
```
Comment (Maffooch) [inline dojo/finding/views.py:2274]: Please delete

File: dojo/finding/views.py [hunk 4]
```diff
@@ -1690,113 +2203,151 @@ class Original(ImageSpec):
...
                     if finding_to_merge_into.dynamic_finding:
                         dynamic = finding.dynamic_finding
 
-                    if finding_to_merge_into.line is None:
-                        line = finding_to_merge_into.line
+                    # code no used
+                    # if finding_to_merge_into.line is None:
+                    #     line = finding_to_merge_into.line
+                    # if finding_to_merge_into.file_path is None:
+                    #     file_path = finding_to_merge_into.file_path
```
Comment (Maffooch) [inline dojo/finding/views.py:2345]: Please delete

**Review by Maffooch — CHANGES_REQUESTED** [review]
I really liked the approach of commenting out things that are unused. Really made reviewing everything easier. Most of these are comments are asking that those comments be deleted

File: dojo/finding/views.py [hunk 5]
```diff
@@ -141,130 +232,209 @@ def findings(request, pid=None, eid=None, view=None, filter_name=None, order_by=
...
-    if 'endpoints' in request.GET:
-        endpoints = request.GET.getlist('endpoints', [])
+    if "endpoints" in request.GET:
+        endpoints = request.GET.getlist("endpoints", [])
         if len(endpoints) == 1:
             endpoint = endpoints[0]
             endpoint = get_object_or_404(Endpoint, id=endpoint)
-            pid = endpoint.product.id
+            # variable 'pid' no used
+            # pid = endpoint.product.id
```
[the remark this answers is not in the document]
Comment (renejal) [reply dojo/finding/views.py:273]: OK

File: dojo/finding/views.py [hunk 6]
```diff
@@ -410,11 +623,12 @@ def close_finding(request, fid):
         missing_note_types = get_missing_mandatory_notetypes(finding)
     else:
         missing_note_types = note_type_activation
-    form = CloseFindingForm(missing_note_types=missing_note_types)
-    if request.method == 'POST':
+    # Unused assignments
+    # form = CloseFindingForm(missing_note_types=missing_note_types)
```
[the remark this answers is not in the document]
Comment (renejal) [reply dojo/finding/views.py:627]: ok

File: dojo/finding/views.py [hunk 7]
```diff
@@ -591,54 +833,57 @@ def reopen_finding(request, fid):
...
 
 
-@user_is_authorized(Finding, Permissions.Finding_Edit, 'fid')
+@user_is_authorized(Finding, Permissions.Finding_Edit, "fid")
 def apply_template_cwe(request, fid):
     finding = get_object_or_404(Finding, id=fid)
 
-    form = FindingFormID(instance=finding)
+    # Unused assignments
+    # form = FindingFormID(instance=finding)
```
[the remark this answers is not in the document]
Comment (renejal) [reply dojo/finding/views.py:857]: OK

File: dojo/finding/views.py [hunk 8]
```diff
@@ -648,143 +893,193 @@ def delete_finding(request, fid):
...
+    )
 
 
-@user_is_authorized(Finding, Permissions.Finding_Edit, 'fid')
+@user_is_authorized(Finding, Permissions.Finding_Edit, "fid")
 def edit_finding(request, fid):
     finding = get_object_or_404(Finding, id=fid)
-    # finding = finding._detag_to_serializable()
-    # finding = finding._retag_to_original()
+    # delete comment
```
[the remark this answers is not in the document]
Comment (renejal) [reply dojo/finding/views.py:985]: ok

File: dojo/finding/views.py [hunk 9]
```diff
@@ -648,143 +893,193 @@ def delete_finding(request, fid):
...
-                       initial={'vulnerability_ids': '\n'.join(finding.vulnerability_ids)})
-    form_error = False
+    form = FindingForm(
+        instance=finding,
+        req_resp=req_resp,
+        can_edit_mitigated_data=finding_helper.can_edit_mitigated_data(request.user),
+        initial={"vulnerability_ids": "\n".join(finding.vulnerability_ids)},
+    )
+    # Unused local variable
+    # form_error = False
```
[the remark this answers is not in the document]
Comment (renejal) [reply dojo/finding/views.py:1001]: ok

File: dojo/finding/views.py (hunk 2 above)
Comment (renejal) [reply dojo/finding/views.py:2162]: ok

File: dojo/finding/views.py (hunk 3 above)
Comment (renejal) [reply dojo/finding/views.py:2274]: ok

File: dojo/finding/views.py (hunk 4 above)
Comment (renejal) [reply dojo/finding/views.py:2345]: ok

File: dojo/finding/views.py [hunk 10]
```diff
@@ -831,56 +1139,76 @@ def edit_finding(request, fid):
...
                 if not new_finding.has_jira_group_issue:
                     if new_finding.has_jira_issue:
-                        jira_issue = new_finding.jira_issue
-
-                        # everything in DD around JIRA integration is based on the internal id of the issue in JIRA
-                        # instead of on the public jira issue key.
-                        # I have no idea why, but it means we have to retrieve the issue from JIRA to get the internal JIRA id.
-                        # we can assume the issue exist, which is already checked in the validation of the jform
+                        # Unsed local variable
+                        # jira_issue = new_finding.jira_issue
```
[the remark this answers is not in the document]
Comment (Maffooch) [reply dojo/finding/views.py:1149]: Got one left over here

File: dojo/finding/views.py [hunk 11]
```diff
@@ -1379,119 +1807,148 @@ def promote_to_finding(request, fid):
...
+                new_jira_issue_key = jform.cleaned_data.get("jira_issue")
                 if new_finding.has_jira_issue:
-                    jira_issue = new_finding.jira_issue
-
-                    # everything in DD around JIRA integration is based on the internal id of the issue in JIRA
-                    # instead of on the public jira issue key.
-                    # I have no idea why, but it means we have to retrieve the issue from JIRA to get the internal JIRA id.
-                    # we can assume the issue exist, which is already checked in the validation of the jform
+                    # vaiable "jira_issue" no used
+                    # jira_issue = new_finding.jira_issue
```
[the remark this answers is not in the document]
Comment (Maffooch) [reply dojo/finding/views.py:1791]: This one too

File: dojo/finding/views.py [hunk 12]
```diff
@@ -1542,143 +1981,169 @@ def apply_cwe_mitigation(apply_to_findings, template, update=True):
...
-                extra_tags='alert-danger')
+                "Unable to delete Template, please revise and try again.",
+                extra_tags="alert-danger",
+            )
     else:
         return HttpResponseForbidden()
 
 
-def download_finding_pic(request, token):
+def download_finding_pic(token):
```
Comment (Maffooch) [inline dojo/finding/views.py:2114]: Okay confirmed that the request object does need to be present because this function is connected to a URL path. Please put the request arg back

https://github.com/DefectDojo/django-DefectDojo/blob/e76dc86e2ce1a53ac00ff46ccdd0e758337ccd4d/dojo/finding/urls.py#L86

Comment (renejal) [reply dojo/finding/views.py]: OK

Comment (renejal) [reply dojo/finding/views.py]: https://github.com/bancolombia/django-DefectDojo/blob/a3c09694810b56ca30eb2db10e52be172783c3ed/dojo/finding/views.py#L2111

**Review by mtesauro — APPROVED** [review]
Approved

**Review by grendel513 — APPROVED** [review]

**Review by grendel513 — APPROVED** [review]

**Review by devGregA — APPROVED** [review]
